pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -116,11 +115,21 @@ func (r *RootSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 	rs := &v1beta1.RootSync{}
 
 	if err := r.client.Get(ctx, rsRef, rs); err != nil {
-		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
 		if apierrors.IsNotFound(err) {
-			r.logger(ctx).Info("Deleting managed objects")
-			return controllerruntime.Result{}, r.deleteClusterRoleBinding(ctx, reconcilerRef)
```

Comment:
Hmm, so this was only deleting the ClusterRoleBinding before?

Reply:
The ClusterRoleBinding was the only one that wasn’t being deleted by the k8s GC.

Reply:
Because it’s not in the same namespace, which is a limitation of ownership.